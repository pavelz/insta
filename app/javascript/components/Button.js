import React, {useState} from 'react'


export default function Button({name = 'Untitled', type='file',title="...",change = {}}) {
    let button = "";
    switch(type){
        case 'button':
            button = (<button className="btn btn-primary">
                {name}
            </button>)
            break;
        case 'file':
            button = (<div className="custom-file"><input onChange={(e) => change(e)} type="file" className="custom-file-input" name={name}/><label className="custom-file-label">{title}</label></div>)
            break;
        case 'submit':
            button = (<input type="submit" className="btn btn-primary" name={name}/>)
            break;
        default:
            button = (<div className="text-danger">Unknown button type</div>)
    }
    return (
       button
    )
}
