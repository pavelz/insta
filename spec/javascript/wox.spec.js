import React from 'react'
import Button from 'Button'
import renderer from 'react-test-renderer'

test('1 + 1 equals 2', () => {
    const component = renderer.create(<Button name="Untitled" title="Hello"/>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});
