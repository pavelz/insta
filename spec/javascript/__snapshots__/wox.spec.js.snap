// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 + 1 equals 2 1`] = `
<div
  className="custom-file"
>
  <input
    className="custom-file-input"
    name="Untitles"
    onChange={[Function]}
    type="file"
  />
  <label
    className="custom-file-label"
  >
    Hello
  </label>
</div>
`;
